### PR TITLE
Github Actions: Add Phpstan

### PR DESCRIPTION
**Pull request by sukhwinder33445**

### CODE REVIEW

File: phpstan-baseline.neon [hunk 1]
```diff
@@ -0,0 +1,1186 @@
...
+			count: 1
+			path: src/Compat/FilterProcessor.php
+
+		-
+			message: "#^Parameter \\#3 \\$subject of function str_replace expects array\\|string, mixed given\\.$#"
+			count: 2
+			path: src/Compat/FilterProcessor.php
+
+		-
+			message: "#^Part \\$column \\(array\\<string\\>\\|string\\) of encapsed string cannot be cast to string\\.$#"
```
Comment (nilmerg) [inline phpstan-baseline.neon:74]: *encapsed*: is this a typo yours or by phpstan devs? :joy:

File: phpstan.neon [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    scanDirectories:
+        - vendor
+
+    universalObjectCratesClasses: # to ignore magic property errors
+        - ipl\Sql\Config
+
+    ignoreErrors:
+        - '#Call to an undefined method ipl\\Sql\\Connection::exec\(\)#'
+        - '#Call to an undefined method ipl\\Sql\\Connection::query\(\)#'
+        - '#Call to an undefined method ipl\\Sql\\Connection::quote\(\)#'
```
Comment (nilmerg) [inline phpstan.neon:23]: Why are these ignored?

File: phpstan.neon [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+        - '#Method .*\\(In|NotIn)::getValue\(\) should return ipl\\Sql\\Select but returns mixed.$#'
```
Comment (nilmerg) [inline phpstan.neon:25]: is this still a thing?

File: src/Filter/InAndNotInUtils.php [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    /**
+     * Set the value of this condition
+     *
+     * @param Select $value
+     *
+     * @return $this
+     */
+    public function setValue($value): self
```
Comment (nilmerg) [inline src/Filter/InAndNotInUtils.php:56]: can the type declaration be used here?

**Review by nilmerg — CHANGES_REQUESTED** [review]
also, please rebase

File: phpstan.neon (hunk 2 above)
Comment (yhabteab) [reply phpstan.neon:23]: magic method calls `__call()`

File: phpstan.neon (hunk 3 above)
Comment (yhabteab) [reply phpstan.neon:25]: Yes!

File: src/Filter/InAndNotInUtils.php (hunk 4 above)
Comment (yhabteab) [reply src/Filter/InAndNotInUtils.php:56]: Argument type or return type declaration?

File: phpstan-baseline.neon (hunk 1 above)
Comment (yhabteab) [reply phpstan-baseline.neon]: phpstan

Comment (yhabteab) [reply phpstan-baseline.neon]: Actually it's not a typo

Comment (nilmerg) [reply phpstan-baseline.neon]: Well, not an unintended typo, yes. But a [non-existent word](https://www.dict.cc/?s=encapsed) anyway.

File: src/Filter/InAndNotInUtils.php (hunk 4 above)
Comment (nilmerg) [reply src/Filter/InAndNotInUtils.php:56]: argument type of course

Comment (yhabteab) [reply src/Filter/InAndNotInUtils.php:56]: No you can't!

```php
( ! ) Fatal error: Declaration of ipl\Sql\Filter\InAndNotInUtils::setValue(ipl\Sql\Select $value): ipl\Sql\Filter\In must be compatible with ipl\Stdlib\Filter\Condition::setValue($value) in /usr/local/src/ipl-sql/src/Filter/InAndNotInUtils.php on line 56
```

**Review by nilmerg — APPROVED** [review]

**Pull request merged**